### PR TITLE
Create python-publish.yml

### PR DESCRIPTION
**Pull request by atenagm1375**

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
```
Comment (realamirhe) [inline .github/workflows/python-publish.yml:9]:
```suggestion
name: PymoNNtorch
```

**Review by realamirhe — COMMENTED** [review]
LGTM

File: .github/workflows/python-publish.yml (hunk 1 above)
Comment (realamirhe) [inline .github/workflows/python-publish.yml:9]:
```suggestion
name: Upload PymoNNtorch Package
```

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+    - name: Build package
+      run: python -m build
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (realamirhe) [inline .github/workflows/python-publish.yml:39]: Please make sure this secret is available in the GitHub secrets section
I couldn't find one in the [settings](https://github.com/cnrl/PymoNNtorch/settings/variables/actions)

**Review by saeedark — APPROVED** [review]

File: .github/workflows/python-publish.yml (hunk 2 above)
Comment (realamirhe) [reply .github/workflows/python-publish.yml:39]: I put @saeedark 's API token, seems we are ready for shipping 🚀

**Review by realamirhe — APPROVED** [review]

File: pymonntorch/NetworkBehavior/Recorder/Recorder.py [hunk 3]
```diff
@@ -58,9 +58,11 @@ def __init__(
     def initialize(self, object):
         super().initialize(object)
 
-        variables = self.parameter("variables", [])
-        if variables == []:
-            variables = self.parameter("arg_0", [])
+        variables = self.parameter("variables", None)
+        if variables is None:
+            variables = self.parameter("arg_0", None)
+            if variables is None:
```
Comment (realamirhe) [inline pymonntorch/NetworkBehavior/Recorder/Recorder.py:64]: seems redundant!

Comment (saeedark) [reply pymonntorch/NetworkBehavior/Recorder/Recorder.py:64]: It's not the best way to handle this, but it's good enough for now.
Keep in mind that the `parameter` function applies the default value type (if not `None`) on values that it returns, therefore it turns a string into a list of its letters which is not the intended value.

**Pull request merged**